spec/ssh_scan/string_ext_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rspec'
+require 'pathname'
+require 'string_ext'
+require 'resolv'
+
+describe String do
+  context "when unhexing a string" do
+    testing_sample = "48656C6C6F20576F726C6421"
+    it "should load all the attributes properly" do
+      expected_unhexedstr = "Hello World!"
+      test_result = testing_sample.unhexify
+      expect(test_result).to eql(expected_unhexedstr)
+    end
+  end
+
+	context "when hexing a string" do
+	testing_sample = "Hello World!"
+		it "should load all the attributes properly" do
+		  expected_hexedstr = "48656c6c6f20576f726c6421"
+		  test_result = testing_sample.hexify
+		  expect(test_result).to eql(expected_hexedstr)
+		end
+	end
+
+	context "when verifying an IP address" do
+	testing_target = "127.0.0.1"
+		it "should load all the attributes properly" do
+		  expected_result = true
+		  test_result = testing_target.ip_addr?
+		  expect(test_result).to eql(true)
+		end
+	end
+
+	context "when resolving a DNS name as IPv4" do
+	testing_dns = "192.168.1.1"
+		it "should load all the attributes properly" do
+		  test_result = testing_dns.resolve_fqdn_as_ipv4.to_s
+		  expect(test_result).to eql("192.168.1.1")
+		end
+	end
```

Comment:
maybe we check github.com and change the test expectation to match on an ip regex?